tests/suite/test_transport_server_tcp_load_balance.py

```diff
@@ -249,3 +251,69 @@ def test_tcp_request_load_balanced_missing_service(
             client.close()
 
         self.restore_ts(kube_apis, transport_server_setup)
+
+    def make_holding_connection(self, host, port):
+        print(f"sending tcp requests to: {host}:{port}")
+        client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client.connect((host, port))
+        client.sendall(b'hold')
+        response = client.recv(4096)
+        endpoint = response.decode()
+        print(f'response: {endpoint}')
+        return client
+
+    def test_tcp_request_max_connections(
```

Comment:
Although adding a test which test the functionality is great but I think the one which checks the key, value in configurtion i.e. stream-conf should take precedence. What do you thing @soneillf5 @pleshakov ?

Reply:
I think we can add a test for that. If the max connections functionality starts to fail, it will help diagnose the problem.

Reply:
I've added a test for the config itself.